src/mhw_lookup.cpp

```diff
@@ -110,12 +126,39 @@ namespace {
 		if(!std::regex_match(realid, IncludeMonsterIdRegex))
 			return false;
 		m.used = true;
+
+		m.hp_total = mb.read_mem<float>(hcompaddr + offsets::MonsterHealthComponent::MaxHealth, true);
+		m.hp_current = mb.read_mem<float>(hcompaddr + offsets::MonsterHealthComponent::CurrentHealth, true);
+		float size_scale = mb.read_mem<float>(maddr + offsets::Monster::MonsterSizeScale, true);
+		float scale_modifier = mb.read_mem<float>(maddr + offsets::Monster::MonsterScaleModifier, true);
+		if (scale_modifier <= 0 || scale_modifier >= 2 ){
+                scale_modifier = 1;
+        }
+
 		// if with SmartHunter we should do the lookup based on the
 		// string id, with info gotten from HunterPie, it's better
 		// to use the numerical id
-		m.name = get_monster_name(numid);
-		m.hp_total = mb.read_mem<float>(hcompaddr + offsets::MonsterHealthComponent::MaxHealth, true);
-		m.hp_current = mb.read_mem<float>(hcompaddr + offsets::MonsterHealthComponent::CurrentHealth, true);
+		const mhw_lookup::monster_data* m_stored_data = get_monster_stored_data(numid);
+		if (m_stored_data) {
+			m.name = m_stored_data->name;
+
+			float modified_size_scale = round(size_scale/scale_modifier*100)/100;
+			m.body_size = m_stored_data->base_size * modified_size_scale;
+
+			const mhw_lookup::crown_preset_data* m_crown_preset = get_crown_preset(m_stored_data->crown_preset);
+			if (m_crown_preset) {
+				if (modified_size_scale <= m_crown_preset->mini) {
+					m.crown = "Mini";
+				}				
+				else if (modified_size_scale >= m_crown_preset->gold) {
+					m.crown = "Gold";				
+				}
+				else if (modified_size_scale >= m_crown_preset->silver) {
+					m.crown = "Silver";
+				}
+			}
+		}	
+		
```

Comment:
Please manage the case when `m_stored_data == 0` - i.e. setting some default values.

Reply:
I set some default values for  `m.name` , `m.body_size` and `m.crown` in the struct from file src/ui.cpp.
Thus, when `m_stored_data == 0 `  these default values would stay. 

```cpp
struct monster_info {
	bool		used = false;
	const char*	name = "<N/A>";
	float		hp_total = -1.0,
			hp_current = -1.0,
			body_size = 0;
	const char*	crown = "";
};
```

Reply:
Ok